tests/granulate_utils/config_feeder/fixtures/api.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from typing import Any, Callable, ContextManager, List, Tuple
+from unittest.mock import patch
+
+from requests_mock.mocker import Mocker
+from requests_mock.request import _RequestObjectProxy
+
+from granulate_utils.config_feeder.client.client import DEFAULT_API_SERVER_ADDRESS
+from granulate_utils.config_feeder.core.models.node import NodeInfo
+
+
+class ApiMock:
+    def __init__(self, **kwargs: Any):
+        self.kwargs = kwargs
+        self._api_mock: Mocker | None = None
+
+    @property
+    def requests(self) -> defaultdict[str, List[_RequestObjectProxy]]:
```

Comment:
You're using internal (private) class, no other more elegant way don't it in this library?

Reply:
Library was not written with types in mind. When request_history and types were added `_RequestObjectProxy` should have been aliased not to give impression that's internal.